terraso_backend/apps/auth/services.py

```diff
@@ -58,18 +59,24 @@ def _persist_user(self, email, first_name="", last_name="", profile_image_url=No
         if last_name:
             user.last_name = last_name
 
-        user_id = str(user.id)
-        try:
-            if not user.profile_image and profile_image_url:
-                user.profile_image = profile_image_service.upload_url(user_id, profile_image_url)
-        except Exception:
-            logger.exception("Failed to upload profile image. User ID: {}".format(user_id))
-
         if update_name:
             user.save()
 
         return user
 
+    def _update_profile_image(self, user, profile_image_url):
+        if not profile_image_url:
```

Comment:
question: won't this exit early with a missing/blank URL and prevent it from updating?

Reply:
Yes, and it should. Since the following logins after the initial one the image is not provided by Google, if we update the URL with the empty value, we are going to erase the value on every login.